pkg/apis/aws/helper/error_codes.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper
+
+import (
+	"regexp"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+var (
+	unauthenticatedRegexp               = regexp.MustCompile(`(?i)(InvalidAuthenticationTokenTenant|AuthFailure|InvalidAccessKeyId|InvalidSecretAccessKey|InvalidSubscriptionId)`)
+	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|SignatureDoesNotMatch|AuthorizationFailed|UnauthorizedOperation|AccessDenied|OperationNotAllowed|Error 403|SERVICE_ACCOUNT_ACCESS_DENIED)`)
+	quotaExceededRegexp                 = regexp.MustCompile(`(?i)((?:^|[^t]|(?:[^s]|^)t|(?:[^e]|^)st|(?:[^u]|^)est|(?:[^q]|^)uest|(?:[^e]|^)quest|(?:[^r]|^)equest)LimitExceeded|Quotas|Quota.*exceeded|exceeded quota|Quota has been met|QUOTA_EXCEEDED|Maximum number of ports exceeded|ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS|VolumeSizeExceedsAvailableQuota)`)
+	rateLimitsExceededRegexp            = regexp.MustCompile(`(?i)(RequestLimitExceeded|Throttling|Too many requests)`)
+	dependenciesRegexp                  = regexp.MustCompile(`(?i)(PendingVerification|Access Not Configured|accessNotConfigured|DependencyViolation|OptInRequired|DeleteConflict|Conflict|inactive billing state|ReadOnlyDisabledSubscription|is already being used|InUseSubnetCannotBeDeleted|VnetInUse|InUseRouteTableCannotBeDeleted|timeout while waiting for state to become|InvalidCidrBlock|already busy for|InsufficientFreeAddressesInSubnet|InternalServerError|internalerror|internal server error|A resource with the ID|VnetAddressSpaceCannotChangeDueToPeerings|InternalBillingError|There are not enough hosts available)`)
```

Comment:
Is the list of error codes filtered to contain only AWS specific error strings? Because I see non-AWS error strings.

Reply:
`InvalidSubscriptionId` - subscription is Azure specific term, hence it could only occur for Azure Shoots
`SERVICE_ACCOUNT_ACCESS_DENIED` - checking https://github.com/gardener/gardener/issues/5364, it could occur only for GCP Shoots
`ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS` - checking our internal ops-guide, it is GCP specific
`VolumeSizeExceedsAvailableQuota` - checking our internal ops-guide, it is OpenStack specific
`ReadOnlyDisabledSubscription` - it could occur only for Azure as subscription is Azure specific term
`VnetInUse` - vnet is only Azure specific term
`VnetAddressSpaceCannotChangeDueToPeerings` - vnet is only Azure specific term
`SkuNotAvailable` - SKU is Azure specific term

These are the ones that I noticed at first look.
You can check the error messages using:
- the git history - maybe the PR that added the error string reveals hints on which infra it occurred
- the internal ops-guide
- the Shoots on our landscapes

There are also a lot gardener specific ones such as `KubeletHasInsufficientMemory`, `KubeletHasDiskPressure`, `KubeletHasInsufficientPID`. What we do about them - do we duplicate them on every extension?

Reply:
I have updated the list.
>There are also a lot gardener specific ones such as KubeletHasInsufficientMemory, KubeletHasDiskPressure, KubeletHasInsufficientPID. What we do about them - do we duplicate them on every extension?

Yes, Otherwise there will be a need to handle it again in gardener/gardener.

Reply:
But doesn't it make sense to handle the kubelet errors centrally rather than in the extensions ? Does it really help if we duplicate them now ?

Reply:
I have dropped kublet specific error codes. PTAL.